src/sys/dune

```diff
@@ -0,0 +1,22 @@
+;; Test of the Sys library
+
+;; this prevents the tests from running on a default build
+
+(alias
+ (name default)
+ (package multicoretests)
+ (deps stm_tests.exe lin_tests.exe lin_tests_dsl.exe))
+
+(executable
+ (name stm_tests)
+ (modules stm_tests)
+ (libraries STM)
+ (preprocess
+  (pps ppx_deriving.show ppx_deriving.eq)))
```

Comment:
Here you don't need to derive an `equal` function as you are only `[@@deriving show ...]` in the code:
```
(preprocess (pps ppx_deriving.show)))
```
Nicolas spotted and fixed many places where we do so yesterday in #132